### DISCUSSION TITLE
Useful new assertion? assertFileDownloadedContains

### DISCUSSION THREAD

**Discussion by colinmackinlay** [livewire/livewire#4981] [3. Feature ideas]
The existing method `assertFileDownloaded` has an undocumented argument to check the entire downloaded file content.

This adds an assertion `assertFileDownloadedContains` to check for some specific partial content in the downloaded file.

```
    public function assertFileDownloadedContains($content)
    {
        $downloadedContent = data_get($this->lastResponse, 'original.effects.download.content');

        PHPUnit::assertStringContainsString(
            $content,
            base64_decode($downloadedContent)
        );

        return $this;
    }
```

If useful I will write tests and add to documentation

**Comment by colinmackinlay** [comment]
I've also found this `assertFileDownloadedNotContains` useful too:
```
public function assertFileDownloadedNotContains($content)
{
    $downloadedContent = data_get($this->lastResponse, 'original.effects.download.content');

    PHPUnit::assertStringNotContainsString(
        $content,
        base64_decode($downloadedContent)
    );

    return $this;
}
```

My main use case is in checking the output of a `print` to PDF function. I can't easily check the contents of a rendered PDF (Spatie Browsershot) but in the testing environment I return the rendered html rather than the PDF conversion. Checking that content helps validate that my print rendering is working.

**Comment by joshhanley** [comment]
@colinmackinlay PR's are welcome! 🙂

**Reply by colinmackinlay** [reply]
PR finally done #5192 :)
